worker/provisioner/provisioner_task.go

```diff
@@ -963,22 +972,24 @@ func (task *provisionerTask) setupToStartMachine(machine *apiprovisioner.Machine
 		arch,
 	)
 	if err != nil {
-		return nil, environs.StartInstanceParams{}, task.setErrorStatus("cannot find tools for machine %q: %v", machine, err)
+		task.setErrorStatus("cannot find tools for machine %q: %v", machine, err)
+		return nil, environs.StartInstanceParams{}, err
 	}
 
-	// ToDo ProvisionerParallelization 2017-10-03
+	// TODO ProvisionerParallelization 2017-10-03
 	// Fix StartInstanceParams struct to carry the availability zone,
 	// overwrite the placement as done currently.
 	// DistributionGroups() can be removed.
-	startInstanceParams, err := constructStartInstanceParams(
+	startInstanceParams, err := task.constructStartInstanceParams(
 		task.controllerUUID,
 		machine,
 		instanceCfg,
 		pInfo,
 		possibleTools,
 	)
 	if err != nil {
-		return nil, environs.StartInstanceParams{}, task.setErrorStatus("cannot construct params for machine %q: %v", machine, err)
+		task.setErrorStatusAndRemoveMachineAZ("cannot construct params for machine %q: %v", machine, err)
```

Comment:
just annotate and return the error, letting startMachines set status/remove from map -- same for other instances of setErrorStatusAndRemoveMachineAZ/setErrorStatus in this method

Reply:
can we leave this one be for now? and do in refinements?

Reply:
yup